### ISSUE TITLE
failing CI

### ISSUE THREAD

**Issue by jreback** [ibis-project/ibis#2181]
seems running out of space on the CI
though recent pushes were ok

**Comment by jreback** [comment]
cc @xmnlab

**Comment by xmnlab** [comment]
@jreback I can work on that today at the end of the day (some deadline here today of another project)

**Comment by datapythonista** [comment]
I guess this is the cause for this error: https://dev.azure.com/ibis-project/ibis/_build/results?buildId=2306&view=logs&j=8f09edc2-e3b7-52de-126a-0225c4f3efa1&t=ee6b35f3-eb84-567b-2649-3343bed41b12&l=1962

**Comment by xmnlab** [comment]
windows jobs are also failing: `Windows fatal exception: access violation` https://dev.azure.com/ibis-project/ibis/_build/results?buildId=2323&view=logs&j=a729dcb5-12ec-5e84-56cf-ed9028c16680&t=01bf3f77-bc94-5d91-05c1-58d17e42c42d&l=782

**Comment by datapythonista** [comment]
@jreback looks like the CI is failing with 3 independent errors:

- **Windows** with an `access violation` error, which seems to be caused by new `arrow-cpp` conda builds (there are new builds of the same 0.16.0 version, not sure why). I'm checking in #2190 if this hypothesis is correct, and a pin to the specific builds in Windows fixes the error for now.

- The **docs, build and benchmarks** builds are failing for the 1 hours timeout of the build. Apparently, conda takes longer to resolve the environment than that. :\ I think the solutions are to use mamba, or pin more dependencies to remove candidates to be considered by the solver (would start by pinning `numpy` to a specific version if we do that).

- The **disk space** error on Linux. @xmnlab is working on this on in #2182, I'm coordinating with him.

In #2189 I introduced an empty commit, and all the errors can be reproduced. Tried reverting up to 18 commits to master, and all errors were also reproduced. So none of them seems to be caused by a recent change in Ibis.

Any thoughts?
